### DISCUSSION TITLE
Running scripts that need other libs installed

### DISCUSSION THREAD

**Discussion by backslashV** [custom-components/pyscript#177] [Q&A]
Can I run python scripts that depend on some external libraries (like selenium, BeautifulSoup, and chromium) using pyscript? currently I'm running the script on my rPi and just sending the result to HA. I'd like to run it locally.

**Comment by craigbarratt** [comment]
I'm not sure I fully understand your question.  Pyscript can import external packages, but those packages need to be installed (eg, via `pip`) on the same host or container that HA is running in.  So in your case those packages would need to be installed on the rPi.

**Reply by backslashV** [reply]
Thank you for your response. Installing packages is exactly the problem as HA terminal doesn't recognize the pip command.

**Comment by craigbarratt** [comment]
I haven't tested this, but try putting your external package requirements in the file `config/pyscript/requirements.txt`.  Just list the packages one per line (you can also specify the version too with `package_name==version`).  Pyscript should then install those packages if they are not already available.

**Reply by backslashV** [reply]
Thanks. It can't find webdriver from the _selenium_ lib.
`Exception in </config/pyscript/scraper_linux.py> line 2: from selenium import webdriver ^ AttributeError: module 'selenium' has no attribute 'webdriver'`
Also, how do I install the _chromium-browser_? I also need the _chromedriver_. Where can I put the binaries for the driver and what path should I use in the script?

**Comment by Chr1stian** [comment]
I am trying to do the same and getting the same errors regarding the selenium webdriver. Placed "selenium" in requirements.txt
`Exception in </config/pyscript/login_startcharging.py> line 1: from selenium import webdriver ^ AttributeError: module 'selenium' has no attribute 'webdriver'`

**Comment by dlashua** [comment]
The issue here is the selenium also requires the installation of a web driver, which is not installable via pip. `requirements.txt` only handles pip installations. So you'll need to find some other way to install the driver on your HASS instance, which may be difficult and is mostly undocumented depending on how you've installed HASS.

**Comment by awb715** [comment]
Any luck here?

I am trying to build an application / scraper that relies on being logged into a certain website. I see Selenium as an option to allow the user to manually login to allow the scrpaing to take place but I would need to automate the download of the driver. Maybe something like this?

https://www.codegrepper.com/code-examples/python/how+to+automate+downloading+a+file+from+a+website+using+python

**Reply by backslashV** [reply]
Nope. Gave up on it.
